sync/src/types/mod.rs

```diff
@@ -958,25 +958,19 @@ impl HeaderView {
         self.inner
     }
 
-    pub fn build_skip<F, G>(
-        &mut self,
-        tip_number: BlockNumber,
-        mut get_header_view: F,
-        fast_scanner: G,
-    ) where
+    pub fn build_skip<F, G>(&mut self, tip_number: BlockNumber, get_header_view: F, fast_scanner: G)
+    where
         F: FnMut(&Byte32, Option<bool>) -> Option<HeaderView>,
         G: Fn(BlockNumber, &HeaderView) -> Option<HeaderView>,
     {
-        let store_first = self.number() <= tip_number;
-        self.skip_hash = get_header_view(&self.parent_hash(), Some(store_first))
-            .and_then(|parent| {
-                parent.get_ancestor(
-                    tip_number,
-                    get_skip_height(self.number()),
-                    get_header_view,
-                    fast_scanner,
-                )
-            })
+        self.skip_hash = self
+            .clone()
```

Comment:
There is no difference between this code and the original code in terms of function. The main change here is that since the header map introduce backend db, `get_header_view` may not just be a cheap memory operation, but  once IO operation, and the reason why the original code does not use get_ancestor directly, but gets the parent first, I guess it is just because get_ancestor is a method that consume self, so here just clone self also works , get the parent does not have any practical significance.

Reply:
There is no difference between this code and the original code in terms of function. The main change here is that since the header map introduce backend db, `get_header_view` may not just be a cheap memory operation, but  once IO operation, and the reason why the original code does not use get_ancestor directly, but gets the parent first, I guess it is just because get_ancestor is a method that consume self, so here just clone self also works , get the parent does not have any practical significance.